torch/_prims/__init__.py

```diff
@@ -1384,12 +1384,17 @@ def _collapsed_shape(shape: ShapeType, start: int, end: int) -> tuple[int, ...]:
     return shape[0:start] + (dim_length,) + shape[end + 1 :]
 
 
+# If must_be_valid is not None, do a torch._check that the collapse is valid else
+# return None, None.
+# When must_be_valid is False, this function returns None when validity is not decided.
+# When must_be_valid is True, this function generates a runtime assertion for validity
+# when it is not decided.
 def _collapse_view_helper(
-    a: TensorLikeType, start: int, end: int
+    a: TensorLikeType, start: int, end: int, must_be_valid: Optional[str]
```

Comment:
Is there a case where this function can return (None, None) that isn't one where must_be_valid would have triggered? i.e. Why add must_be_valid instead of just checking for None at the caller?

Reply:
because we want the sympy expressions that we should convert to runtime assertions. 
the alternative to not passing must_be_valid is returning a 3rd output being valid_op variable when we return None, None,
make sense?